### PR TITLE
gets applicant and locks the row to the user for an hour

### PR DESCRIPTION
**Pull request by klanmiko**
we need an update trigger that gets set/created on application load. There is also preliminary work in loading the model from the columns during application load. 

In further commits I will map an enum in a separate table that will be loaded into the Model in the same process, which will be more useful and is declarative enough for the frontend to render it.

Will add state to load_csv in future commits to do annotation of column type when applications are first loaded.

Need to do a whole lot more work for admin functionality and computation of final results

### CODE REVIEW

File: src/db/models/ApplicationsModel.py [hunk 1]
```diff
@@ -38,6 +53,60 @@ def store(self, row):
...
+        results = cursor.fetchone()
+        return results
+    
+    def getNewLockedApplicantRow(self, user):
+        columns_to_return = list(ApplicationsModel.Model.keys())
+        columns_to_return.remove('user_editing')
+
+        conn = connection.get_connection()
+
+        query = sql.SQL("""SELECT {} FROM {} WHERE "score"=0 AND ("user_editing" is null  OR now() - "last_modified" > interval '2 hours') LIMIT 1 FOR UPDATE""").format(
```
[the remark this answers is not in the document]
Comment (thomasmunduchira) [reply src/db/models/ApplicationsModel.py]: Is the plan to score an applicant from 1 to 5? Just making sure that 0 isn't a a valid score

File: src/db/models/ApplicationsModel.py [hunk 2]
```diff
@@ -38,6 +53,60 @@ def store(self, row):
         )
         connection.execute_query(query, values)
 
+    def getLockedApplicantRow(self, user):
+        columns_to_return = list(ApplicationsModel.Model.keys())
+        columns_to_return.remove('user_editing')
+
+        query = sql.SQL("""SELECT {} FROM {} WHERE "user_editing"={} AND now() - last_modified < interval '2 hours'""").format(
```
[the remark this answers is not in the document]
Comment (thomasmunduchira) [reply src/db/models/ApplicationsModel.py]: Can we do 1 hour? I don't see the need to make the timeout that long

File: src/db/models/ApplicationsModel.py [hunk 3]
```diff
@@ -38,6 +53,60 @@ def store(self, row):
...
+            sql.SQL(', ').join([sql.Identifier(column) for column in columns_to_return]),
+            sql.Identifier(ApplicationsModel.TableName),
+            sql.Placeholder()
+        )
+
+        cursor = connection.execute_query(query, (user, ))
```
[the remark this answers is not in the document]
Comment (klanmiko) [reply src/db/models/ApplicationsModel.py]: bruh it's a tuple

File: src/db/models/ApplicationsModel.py [hunk 4]
```diff
@@ -38,6 +53,60 @@ def store(self, row):
         )
         connection.execute_query(query, values)
 
+    def getLockedApplicantRow(self, user):
```
[the remark this answers is not in the document]
Comment (klanmiko) [reply src/db/models/ApplicationsModel.py]: single row so `getLockedApplicantRowForUser`

File: src/db/models/ApplicationsModel.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (thomasmunduchira) [reply src/db/models/ApplicationsModel.py]: Oh whoops I missed that

File: src/controller/review.py [hunk 5]
```diff
@@ -1,3 +1,37 @@
...
+def doCSV():
+    with open(os.path.join(_dirname, "../lib/sample.csv"), encoding="utf-8") as data:
+        loadCSV.upload_csv(data)
+    return Response(status=200)
+
+@review.route('/review')
+def getApplicant():
+    model = ApplicationsModel.get_applications_model()
+    try:
+        data = model.getLockedApplicantRowForUser(1)
```
Comment (thomasmunduchira) [inline src/controller/review.py]: Putting a note here that the user is hardcoded since authentication is not done. This should be fixed in a future PR

**Pull request merged**